### PR TITLE
Show errors in the pattern as well as the text

### PR DESCRIPTION
**Pull request by ltratt**
This PR ultimately moves things to the point that you get output such as:

```
Pattern (error at line 4):
    1: a
    2: b
    3: c
>>  4: d

Text (error at line 4):
    1: a
    2: b
    3: c
>>  4: z
```

Notice that we stop printing output after we've hit the "mark" (since it's not useful and just causes the console to scroll endlessly). I'm also no longer entirely sure if it's useful to print the line numbers out, given that we tell users the line elsewhere *and* show a mark. Thoughts?

### CODE REVIEW

**Comment by vext01** [comment]
> I'm also no longer entirely sure if it's useful to print the line numbers out, given that we tell users the line elsewhere and show a mark.

You might be right.

Naively, that'd look like:
```
Pattern (error at line 4):
    a
    b
    c
>>  d

Text (error at line 4):
    a
    b
    c
>>  z
```

But I somehow miss the left margin indicating where the line starts proper, so how about:
```
Pattern (error at line 4):
  |a
  |b
  |c
>>|d

Text (error at line 4):
  |a
  |b
  |c
>>|z
```

You know what else could be useful to understand the lack of backtracing? Another marker that shows where the first matched line was?

**Comment by ltratt** [comment]
Addressed in https://github.com/softdevteam/fm/pull/12/commits/869c7cad78cb692611fd58c7b2f592ce69ca560f.

**Comment by ltratt** [comment]
Note also the subtly changed output here:

```
        assert_eq!(
            helper("a\n", "a\n\nb"),
            "Pattern (error at line 2):
   |a
>>

Text (error at line 3):
   |a
   |
>> |b
"
        );
```

I prefer this as it makes clearer (to me at least) that the pattern has been completely matched (note that `>>` has no `|` marker) but there is still unmatched text.

**Comment by vext01** [comment]
> note that >> has no | marker

yes, I like this.

File: src/lib.rs [hunk 1]
```diff
@@ -246,24 +250,29 @@ impl<'a> FMatcher<'a> {
...
                                 });
                             }
                         }
                         return Ok(());
                     } else {
-                        if self.skip_blank_lines(&mut ptn_lines, Some(x)).0.is_none() {
-                            return Ok(());
+                        match self.skip_blank_lines(&mut ptn_lines, Some(x)) {
+                            (Some(_), skipped) => {
+                                return Err(FMatchError {
```
Comment (vext01) [inline src/lib.rs:268]: I was supposed to do this myself, but I think the code to make the error is long enough and appears frequently enough, that we should add a `::new()` method to make things more concise. What do you think?

File: src/lib.rs [hunk 2]
```diff
@@ -431,23 +431,31 @@ impl fmt::Display for FMatchError {
...
-            mark_line: Option<usize>,
+            mark_line: usize,
         ) -> fmt::Result {
-            for (num, line) in s.lines().enumerate() {
-                let mark = match mark_line {
-                    Some(ml) if ml == num + 1 => ">> ",
-                    _ => "   ",
-                };
-                writeln!(f, "{}{:width$}: {}", mark, num + 1, line, width = lno_chars)?;
+            let mut num_lines = 0;
```
Comment (vext01) [inline src/lib.rs]: isn't `num_lines` a mirror of `i` here?

**Comment by vext01** [comment]
Just a couple of small comments.

File: src/lib.rs (hunk 1 above)
Comment (ltratt) [reply src/lib.rs:268]: I'm neutral. Can be done as a separate PR?

Comment (vext01) [reply src/lib.rs:268]: Sure.

File: src/lib.rs (hunk 2 above)
Comment (ltratt) [reply src/lib.rs]: Ha, sort of, yes! Fixed in https://github.com/softdevteam/fm/pull/12/commits/7836a8b3aef84429fe5ccccbca9ca9a8b70d83f7.

**Comment by vext01** [comment]
That's better. Please squash.

**Comment by ltratt** [comment]
Hmm, actually, squashing doesn't make sense here. But I should reword that last commit and force push if you're OK with that?

**Comment by vext01** [comment]
As you wish.

**Comment by ltratt** [comment]
Force pushed.

**Pull request merged**